pkg/resource/deploy/step.go

```diff
@@ -942,32 +932,48 @@ func (s *ImportStep) Apply(preview bool) (resource.Status, StepCompleteFunc, err
 		}
 
 		if len(inputProperties) == 0 {
-			logging.V(9).Infof("Importing %v with required properties", s.URN())
-
-			for _, p := range r.InputProperties {
-				if p.IsRequired() {
-					k := resource.PropertyKey(p.Name)
-					s.new.Inputs[k] = s.old.Inputs[k]
-				}
-			}
+			logging.V(9).Infof("Importing %v with all properties", s.URN())
+			s.new.Inputs = s.old.Inputs.Copy()
 		} else {
 			logging.V(9).Infof("Importing %v with supplied properties: %v", s.URN(), inputProperties)
-
 			for _, p := range inputProperties {
 				k := resource.PropertyKey(p)
 				if value, has := s.old.Inputs[k]; has {
 					s.new.Inputs[k] = value
 				}
 			}
 		}
-	} else {
-		// Set inputs back to their old values (if any) for any "ignored" properties
-		processedInputs, res := processIgnoreChanges(s.new.Inputs, s.old.Inputs, s.ignoreChanges)
-		if res != nil {
-			return resource.StatusOK, nil, res.Error()
+
+		// Check the provider inputs for consistency. If the inputs fail validation, the import will still succeed, but
+		// we will display the validation failures and a message informing the user that the failures are almost
+		// definitely a provider bug.
+		_, failures, err := prov.Check(s.new.URN, s.old.Inputs, s.new.Inputs, preview, s.new.SequenceNumber)
+		if err != nil {
+			return rst, nil, err
+		}
+
+		if issueCheckFailures(s.deployment.Diag().Warningf, s.new, s.new.URN, failures) {
+			ref, err := providers.ParseReference(s.Provider())
+			contract.Assert(err == nil)
+
+			pkgName := ref.URN().Type().Name()
+			s.deployment.Diag().Warningf(diag.Message(s.new.URN,
+				"One or more imported inputs failed to validate. "+
+					"This is almost certainly a bug in the `%s` provider. "+
+					"The import will still proceed, but you will need to edit the generated code after copying it into your program"),
```

Comment:
cc @lukehoban @t0yv0 @mikhailshilkov for the wording here. Though validation failures are pretty much provider bugs by definition, there may be situations in which the provider cannot do any better, and filing an issue may be a waste of time. The rest of the message looks great to me.

Reply:
Yeah I don't feel confident that the statement "This is almost certainly a bug in the `%s` provider" is true. Were all previous import errors bugs in providers? Or do we think this PR would fix them all?

Reply:
Not exactly. After these changes, though, it is definitively the responsibility of the provider to provide the most accurate set of inputs it can from `Read`, so any difference that e.g. causes a validation failure would be a problem in the provider.